Allows developers to add their own filters to the active filters list

webdados opened the pull request:
Fixes #36704

### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes #36704 allowing developers to add their own custom filters to the active filters widget

<!-- The next section is mandatory. If your PR doesn't require testing, please indicate that you are purposefully omitting instructions. -->

- [x] This PR is a very minor change/addition and does not require testing instructions

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

<!-- Mark completed items with an [x] -->

### FOR PR REVIEWER ONLY:

-   [x] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

webdados committed:
Allows developers to add their own filters to the active filters list

Fixes #36704

rrennick commented:
@webdados Sorry for the delay in reviewing. Thanks for submitting this. This looks okay except it is missing doc blocks on the hooks and doesn't have a changelog entry. You can add the changelog entry with

```
pnpm --filter=plugins/woocommerce run changelog add
```

webdados commented:
Sorry, not sure what you mean by "missing doc blocks on the hooks".
Also, any way to add the changelog here, thought the web interface?

rrennick commented:
> Sorry, not sure what you mean by "missing doc blocks on the hooks".

I meant PHPDocs. An [example](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php#L1754:L1762). We follow the [WordPress docs standard](https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#docblock-formatting) for this.

> any way to add the changelog here, thought the web interface?

No, but if you have `composer` set up you can create it using `composer exec -- changelogger` in the `plugins/woocommerce` folder.

rrennick commented:
> any way to add the changelog here, thought the web interface?

@webdados If you add the docblocs on the hooks, I'll add the changelog entry.

webdados committed:
Add params and docblocks to the new hooks 

Add params and docblocks to the new proposed hooks woocommerce_widget_layered_nav_filters_start and woocommerce_widget_layered_nav_filters_end

webdados commented:
> @webdados If you add the docblocs on the hooks, I'll add the changelog entry.

@rrennick Done. I omitted the "since" tag as I have no idea when this is going to be merged.
I also took the liberty to add the `$args` and `$instance` params as the 3rd party developers might have a use for it.

rrennick committed:
Add changelog file

rrennick commented:
> I omitted the "since" tag as I have no idea when this is going to be merged.

I've added the changelog. Could you add them for `7.6.0`, ping me, and I'll merge shortly after?

webdados committed:
Add @since tag to hooks doc

webdados commented:
Done

rrennick reviewed (commented):
Move the changelog to the correct folder.

rrennick committed:
Move the changelog to the correct folder

rrennick reviewed (approved):
@webdados Thanks for your patience with this.
